src/app/components/pool/graphs/heatmap/nodes-heatmap.component.ts

```diff
@@ -341,31 +343,36 @@ export class NodesHeatmapComponent implements AfterViewInit, OnChanges, OnDestro
      */
     private _displayTileTooltip(titleNode) {
         titleNode.text((tile) => {
-            const count = tile.node.runningTasksCount;
+            const taskCount = tile.node.runningTasksCount;
+            const taskSlotsCount = tile.node.runningTaskSlotsCount;
             if (tile.node.state === NodeState.unusable) {
                 return `Error: Unusable state on node (${tile.node.id})`;
             } else if (tile.node.state === NodeState.startTaskFailed) {
                 return `Error: Start task failed on node (${tile.node.id})`;
             } else if (tile.node.state === NodeState.unknown) {
                 return `Error: Unknown state on node (${tile.node.id})`;
             } else {
-                return `${count} tasks running on node (${tile.node.id})`;
+                return `${taskCount} tasks (${taskSlotsCount}/${this.pool.taskSlotsPerNode} slots) running on node (${tile.node.id})`;
             }
        });
     }
 
-    private _getTaskHeight(tileSize: number, node: Node) {
+    private _getTaskSlotsHeight(tileSize: number, node: Node) {
         const taskSlotsPerNode = this.pool.taskSlotsPerNode;
-        const taskHeight = Math.floor(tileSize / taskSlotsPerNode);
+        const taskSlotsCount = node.runningTaskSlotsCount;
+        let taskSlotsHeight = 0;
+        if (taskSlotsPerNode) {
```

Comment:
Do we need this check? I think taskSlotsPerNode will always be defined, but if that's not the case we can default it to 1 since that's the default at the API level.

Reply:
Yeah I assumed taskSlotsPerNode would always be defined too but this check was supposed to be an extra safety check.

Reply:
Removed the extra check